stream.go

```diff
@@ -965,34 +964,25 @@ func (a *csAttempt) finish(err error) {
 	a.mu.Unlock()
 }
 
-func (ac *addrConn) newClientStream(ctx context.Context, desc *StreamDesc, method string, t transport.ClientTransport, opts ...CallOption) (_ ClientStream, err error) {
-	ac.mu.Lock()
-	if ac.transport != t {
-		ac.mu.Unlock()
-		return nil, status.Error(codes.Canceled, "the provided transport is no longer valid to use")
-	}
-	// transition to CONNECTING state when an attempt starts
-	if ac.state != connectivity.Connecting {
```

Comment:
Changing connectivity state when creating a stream on the addrConn in general doesn't sound right.
Also, this is to create a general purpose stream on the addrConn (as opposite to a *health checking* stream), so changing connectivity state is just not going to work.

`TransientFailure<->Ready` is already happening today with the health check stream.

This function is called only when connection is created, so the state is not IDLE.

Reply:
> `TransientFailure<->Ready` is already happening today with the health check stream.

This is not true, because new stream will set ac's state to connecting. The fix is to manually set state to connecting before calling newStream(), in the health checking goroutine.

Reply:
No, the "invalid" transition still happens when the health checking stream is good, but server reports unhealthy. But we still need the manual transition to connecting when stream is broken (this behavior is defined in the spec).